paper/paper.md

```diff
@@ -108,14 +112,23 @@ Accordingly to the architecture, we chose the MNIST dataset [@lecun-bouttou-beng
 ```julia
 using MLDatasets 
 
-# download the dataset without the user intervention
-ENV["DATADEPS_ALWAYS_ACCEPT"] = true  
+function getdata(args; T = Float32) #T for types
+  ENV["DATADEPS_ALWAYS_ACCEPT"] = "true"
+
+  # Loading Dataset	
+  xtrain, ytrain = MLDatasets.MNIST(Tx = T, split = :train)[:]
+  xtest, ytest = MLDatasets.MNIST(Tx = T, split = :test)[:]
 
-xtrn, ytrn = MNIST.traindata(Float32) # MNIST training dataset
-ytrn[ytrn.==0] .= output_classes # re-arrange indices
-xtst, ytst = MNIST.testdata(Float32) # MNIST test dataset
-ytst[ytst.==0] .= output_classes # re-arrange indices
+  # Reshape Data in order to flatten each image into a linear array
+  xtrain = Flux.flatten(xtrain)
+  xtest = Flux.flatten(xtest)
+
+  # One-hot-encode the labels
+  ytrain, ytest = onehotbatch(ytrain, 0:9), onehotbatch(ytest, 0:9)
+  return xtrain, ytrain, xtest, ytest
+end
 
+xtrn, ytrn, xtst, ytst = getdata(args)
```

Comment:
Is this working with KnetNLPModels?

If not, make a function 
```julia
function get_MNIST()
  xtrain, ytrain = MLDatasets.MNIST(Tx = T, split = :train)[:]
  xtest, ytest = MLDatasets.MNIST(Tx = T, split = :test)[:]
end
```

Reply:
my code should work, however the getData make sure we turn the y into onehot encoding as required by the model

Reply:
The KnetNLPModels consider `y` as an integer and not an onehot encoding

Reply:
this is what used before
`ytrn[ytrn.==0] .= output_classes ` this is one hot encoding, all classes are zero, except the current class

Reply:
No  `ytrn[ytrn.==0] .= output_classes` turns every integer from `ytrn` whose value is 0 into `outputclasses` who's valuing 10.

example:
```latex
output_classes = 10 
ytrn = [0,1,5,0]
ytrn[ytrn.==0] .= output_classes
ytrn == [10,1,5,10] #true
```

Reply:
my mistake, 
I changed the code based on your example